pkg/splunk/client/awss3client.go

```diff
@@ -142,7 +156,9 @@ func NewAWSS3Client(ctx context.Context, bucketName string, accessKeyID string,
 		}
 	}
 
-	cl := fn(ctx, region, accessKeyID, secretAccessKey)
+	endpointWithRegion := fmt.Sprintf("%s%s%s", region, awsRegionEndPointDemarcator, endpoint)
```

Comment:
It may be clean to pass the awsConfig as a whole, instead of packing and unpacking again?

Reply:
That would require a lot of changes on the client code in general. Decided to use this method to keep it clean across different cloud providers. 

cc: @vivekr-splunk